site/docs/guides/tls.md

```diff
@@ -0,0 +1,84 @@
+# TLS Ingress with Minikube
+
+In this guide we walk through the process of configuring a Nessie Service with secure HTTPS transport in
+[Minikube](https://minikube.sigs.k8s.io/docs/).
+
+## Setting up Minikube
+
+1. Start Minikube cluster: `minikube start`
+2. Enable NGINX Ingress controller: `minikube addons enable ingress`
+
+## Creating Certificates
+
+If you have your own trusted certificate you can use it to sign a new certificate for the Nessie service.
+
+Otherwise, the rest of this section shows how can create a self-signed certificate and add it the list of trusted
+certificates in the client OS.
+
+To generate a fresh pair of a key and a (self-signed) certificate:
+
+```shell
+openssl req -new -subj "/CN=Nessie" -addext "subjectAltName = DNS:nessie.local" \
+  -newkey rsa:2048 -keyout nessie-key.pem -out nessie.crt -x509 -nodes
+```
+
+Note the `-nodes` option. It is used only for the sake of simplicity of this example deployment. Also, newer `openssl`
+versions deprecated it in favour of `-noenc`.
+
+Add the new certificate to the local set of CA certificates on the client host (where the Nessie clients,
+such as Nessie CLI / `curl` / browser, are going to run).
+
+The following example is for Linux:
+
+1. `sudo cp nessie.crt /usr/local/share/ca-certificates/nessie.crt`
+2. `sudo update-ca-certificates`
+
+This should output something like "Certificate added: CN=Nessie".
+
+## Creating a k8s Secret for the Nessie Certificate
+
+```shell
+kubectl -n nessie-ns create secret tls nessie-tls \
+  --cert=nessie.crt --key=nessie-key.pem
+```
+
+## Deploying Nessie with Helm
+
+```shell
+helm install nessie -n nessie-ns helm/nessie \
+  --set ingress.enabled=true \
+  --set ingress.hosts[0].host='nessie.local' \
+  --set ingress.hosts[0].paths[0]='/' \
+  --set ingress.tls[0].secretName=nessie-tls \
+  --set ingress.tls[0].hosts[0]='nessie.local'
+```
+
+The deployment process may take some time. Use the following command to check its status and get the ingress IP address.
+
+```shell
+kubectl get ingress -n nessie-ns
+```
+
+Add an entry in the local hosts file (e.g. `/etc/hosts`) mapping that IP address to `nessie.local`, for example:
```

Comment:
While this works, you could avoid fiddling with /etc/hosts by using one or both of the methods below:

1. for curl only: use the `--resolve` option, e.g. 

```shell
curl --resolve "nessie.local:443:$(minikube ip)" https://nessie.local/api/v2/config
```

2. for all tools and browsers: use a nip.io hostname, e.g. `nessie.$(minikube ip).nip.io`: such a host name will resolve to whatever IP `minikube ip` indicates.  E.g. 

```shell
ping nessie.$(minikube ip).nip.io
PING nessie.192.168.49.2.nip.io (192.168.49.2): 56 data bytes
64 bytes from 192.168.49.2: icmp_seq=0 ttl=64 time=0.108 ms
```

Then you would use `nessie.192.168.49.2.nip.io` as the hostname for your certificates and ingress rules.

Reply:
Yeah, but the certificate must have the exact host name, which is not known until we start / configure ingress... I guess that would create a chicken and egg problem.

Also, the java client still depends on OS-level host name resolution (although I did not put and example for it).

Reply:
updated